modules/explorer/update.go

```diff
@@ -330,8 +354,8 @@ func dbAddFileContractID(tx *bolt.Tx, id types.FileContractID, txid types.Transa
 	mustPutSet(b, txid)
 }
 func dbRemoveFileContractID(tx *bolt.Tx, id types.FileContractID, txid types.TransactionID) {
-	// TODO: delete bucket when it becomes empty
 	mustDelete(tx.Bucket(bucketFileContractIDs).Bucket(encoding.Marshal(id)), txid)
+	deleteEmptyNestedBucket(tx.Bucket(bucketFileContractIDs), encoding.Marshal(id))
```

Comment:
this name doesn't feel clear enough to me. Perhaps move the conditional to the callsite, i.e.:
```go
func dbRemoveFileContractID(tx *bolt.Tx, id types.FileContractID, txid types.TransactionID) {
    // might be a cleaner way to write this...
    bucket := tx.Bucket(bucketFileContractIDs).Bucket(encoding.Marshal(id))
    mustDelete(bucket, txid)
    if bucketIsEmpty(bucket) {
        tx.Bucket(bucketFileContractIDs).DeleteBucket(encoding.Marshal(id))
    }
}

func bucketIsEmpty(bucket *bolt.Bucket) bool {
    k, _ := bucket.Cursor().First()
    return k == nil
}
```
In general, I prefer having the control flow explicit at the callsite rather than hidden inside helper functions (with `assertNil` being an obvious exception :P)